### PR TITLE
Update posts API functions, add support for files API

### PR DESCRIPTION
**Pull request by adamdejl**

### CODE REVIEW

**Comment by hasali19** [comment]
Hmm, when you commit it's supposed to compile the typescript and update dist/index.js. Did you run `git commit` or do something fancier?

Did you also run `npm install`?

**Comment by adamdejl** [comment]
I forgot about the `npm install`... Is it ok now?

File: src/index.ts [hunk 1]
```diff
@@ -38,19 +45,20 @@ export default class ApiClient {
     summary,
     content,
     tags,
+    files,
   }: Post): Promise<Response<Post>> {
+    let formData = new FormData();
+    formData.append("title", title);
+    if (summary) {
+      formData.append("summary", summary);
+    }
```
Comment (hasali19) [inline src/index.ts]: Is summary required or not, I thought your changes to the backend made it required?

**Review by hasali19 — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (adamdejl) [reply src/index.ts]: Yes, it is mandatory. I was just mindlessly rewriting the original code there.

**Pull request merged**